dist/IO/lib/IO/Socket.pm

```diff
@@ -396,218 +396,468 @@ IO::Socket - Object interface to socket communications
 
 =head1 SYNOPSIS
 
-    use IO::Socket;
+    #!/usr/bin/env perl
+    use strict;
+    use warnings;
+    use feature 'say';
+
+    use IO::Socket qw(AF_INET AF_UNIX SOCK_STREAM SHUT_WR);
+
+    # create a new AF_INET socket
+    my $sock = IO::Socket->new(Domain => AF_INET);
+    # which is the same as
+    $sock = IO::Socket::INET->new();
+
+    # create a new AF_UNIX socket
+    $sock = IO::Socket->new(Domain => AF_UNIX);
+    # which is the same as
+    $sock = IO::Socket::UNIX->new();
+
+    # Let's create a TCP server on localhost:3333
```

Comment:
Perhaps this extended example could go in an EXAMPLES heading? I tend to glaze over when synopses get into long self-contained examples.